api/api.go

```diff
@@ -947,29 +947,58 @@ func (l *VolumeLocator) MergeVolumeSpecLabels(s *VolumeSpec) *VolumeLocator {
 	return l
 }
 
-func (v *Volume) IsPermitted(ctx context.Context) bool {
-	return v.GetSpec().IsPermitted(ctx)
+func (v *Volume) IsPermitted(ctx context.Context, accessType Ownership_AccessType) bool {
+	return v.GetSpec().IsPermitted(ctx, accessType)
 }
 
-func (v *VolumeSpec) IsPermitted(ctx context.Context) bool {
-	if v.GetOwnership() != nil {
-		if userinfo, ok := auth.NewUserInfoFromContext(ctx); ok {
-			// Check Access
-			return v.IsPermittedFromUserInfo(userinfo)
-		} else {
-			// There is no user information in the context so
-			// authorization is not running
-			return true
-		}
+func (v *VolumeSpec) IsPermitted(ctx context.Context, accessType Ownership_AccessType) bool {
+	if v.IsPublic() {
+		return true
 	}
 
-	// There is no ownership on this volume, so allow access
-	return true
+	// Volume is not public, check permission
+	if userinfo, ok := auth.NewUserInfoFromContext(ctx); ok {
+		// Check Access
+		return v.IsPermittedFromUserInfo(userinfo, accessType)
+	} else {
+		// There is no user information in the context so
+		// authorization is not running
+		return true
+	}
 }
 
-func (v *VolumeSpec) IsPermittedFromUserInfo(user *auth.UserInfo) bool {
+func (v *VolumeSpec) IsPermittedFromUserInfo(user *auth.UserInfo, accessType Ownership_AccessType) bool {
+	if v.IsPublic() {
```

Comment:
Redundant with `v.IsPublic` check in `IsPermitted`, but maybe safe to leave in if `IsPermittedFromUserInfo` is called directly? Though `FromUserInfo` implies that it might not check for volume info.

Reply:
> IsPermittedFromUserInfo is called directly?

yeah exactly. since the function is exported, someone can call that without checking first if it is public.